_posts/2018-09-26-artsy-frontend-history.markdown

```diff
@@ -0,0 +1,172 @@
+(emphasis theirs) that they describe is a freedom from tangled jQuery code everywhere. I think our definition of
+freedom on the web frontend has evolved since then, but that's just my feeling.
+
+The other key component to our web frontend was [CoffeeScript][]. According to its documentation, "CoffeeScript is
+a little language that compiles into JavaScript", which was pretty important at the time. JavaScript in 2011 is
+very different from JavaScript today. The CoffeeScript docs also state that "JavaScript has always had a gorgeous
+heart", which I'm not sure I'd agree with to be honest, but the CoffeeScript project really shows how a handful of
+engineers working to improve something they care about can change and entire industry. While I don't think
+contemporary JavaScript would have gotten as good as it has without CoffeeScript, it's a bit anachronistic to see
+it used today.
+
+Our goal as a (very small!) engineering team at the time was to keep our moving parts to a minimum.
+Rails+SASS+CoffeeScript+Backbone helped us achieve that goal, and we couldn't have gotten this far without the help
+of those projects.
+
+## Ezel
+
+In November 2013, we split our web frontend from the API backend. You can read
+[all the details in this blog post](2013_review), but the story is summarized nicely as "moving from a single
+monolithic application to modular Backbone apps that run in Node and the browser and consume our external API."
+This move from monolith to modular systems continues to influence day-to-day work on the Artsy Engineering team.
+
+We moved our API from Rails to [Grape][], partially motivated by a desire to build an iOS application that would
+consume this API. We faced a lot of problems with the split, including SEO problems, severe page load times,
+maintaining duplicated backend and frontend UI templates, slow test suites, and poor developer productivity. We
+took the project of building our mobile web frontend, m.artsy.net (still known as "martsy" internally) as an
+opportunity to address these problems.
+
+We built our new site with [Node.js][node] since it easily allowed server-side rendering. We split out areas of
+concern into separate "apps", with their own bundled CSS/JS to help page load times. We server-side rendered
+above-the-fold content and used client-side JS to load the rest, which helped SEO and user experience. We took a
+[BEM][]-like approach to our CSS, which helped developer productivity. Our technical decisions were driven
+primarily by our desire to create great user experiences.
+
+And because we are an open source by default organization, we collected these approaches into an open source
+project called [Ezel][].
+
+We used Ezel for a few years without too much change in our web frontend stack. It proved really useful for helping
+build other web apps – CMS systems for our partners, auction-management systems for our admins, all kinds of
```

Comment:
This is a little confusing. CMS, Ohm, etc. were build largely on the backs of new internal libs like kinetic and watt so I'm not sure the connection to ezel. (Although maybe you're talking about other apps? 😅)

Reply:
Yeah, I think I was unclear here. It sounds then, that Ezel never really took off?

Reply:
I know of other companies (outside of Artsy) that used it, and I think we used it for some other internal apps (maybe refraction?). But by nature Ezel made it easy to mount many "apps" (pages) inside a single shell, so there wasn't a need to have many separate client projects that were generated using it.